### PR TITLE
[TAS-346] BE fix: do not send question with one answer

### PR DESCRIPTION
**Pull request by axelgcd**

### CODE REVIEW

File: controllers/data.controller.js [hunk 1]
```diff
@@ -256,14 +256,21 @@ exports.sendQuestionToFrontend = async (req, res) => {
...
+        questions[Math.floor(Math.random() * questions.length)];
 
-    const selectedQuestion =
-      questions[Math.floor(Math.random() * questions.length)];
+      multipleChoices = await getOtherMultipleChoiceAnswers(
+        selectedQuestion,
+        currentTier,
+      );
+      tries += 1;
+    }
```
Comment (aknur1999) [inline controllers/data.controller.js:274]: please consider this: If after 5 tries no suitable questions are found, return an error

**Review by aknur1999 — CHANGES_REQUESTED** [review]

File: controllers/data.controller.js [hunk 2]
```diff
@@ -256,14 +258,30 @@ exports.sendQuestionToFrontend = async (req, res) => {
...
+    if (multipleChoices.length < 4) {
```
Comment (aknur1999) [inline controllers/data.controller.js:275]: what is the function of this code..?

**Review by aknur1999 — CHANGES_REQUESTED** [review]

File: controllers/data.controller.js (hunk 2 above)
Comment (aknur1999) [reply controllers/data.controller.js:275]: I think maybe by mistake it was added

Comment (axelgcd) [reply controllers/data.controller.js:275]: > I think maybe by mistake it was added

The purpose is to throw an error of there is no suitable question to send as you ask in previous review

**Review by aknur1999 — APPROVED** [review]

**Pull request merged**